onnx/test/cpp/data_propagation_test.cc

```diff
@@ -197,5 +198,170 @@ agraph (int32[2,5] x) => (int32[1,2] w)
   EXPECT_TRUE(CompareShape(propagated_tsp, expected_tsp));
 }
 
+TEST(DataPropagationImplTest, SizeTest) {
+  const char* code = R"ONNX(
+agraph (int64[1] x) => (int32[1] w)
+<int64[3] init = {2,3,5}>
+{
+    z = Size(init)
+    w = Cast<to = 7>(z)
+}
+)ONNX";
+  TensorShapeProto expected_tsp;
+  expected_tsp.mutable_dim()->Add()->set_dim_value(3);
+  const auto propagated_tsp = RunDataPropagation(code);
+  EXPECT_TRUE(CompareShape(propagated_tsp, expected_tsp));
+}
+
+TEST(DataPropagationImplTest, AddTest) {
+  const char* code = R"ONNX(
+agraph (int32[2,4,5] x, int32[2,4,5] y) => (int32[3] w)
+{
+    xs = Shape(x)
+    ys = Shape(y)
+    z = Add(xs, ys)
+    w = Cast<to = 7>(z)
+}
+)ONNX";
+  TensorShapeProto expected_tsp;
+  expected_tsp.mutable_dim()->Add()->set_dim_value(4);
+  expected_tsp.mutable_dim()->Add()->set_dim_value(8);
+  expected_tsp.mutable_dim()->Add()->set_dim_value(10);
+  const auto propagated_tsp = RunDataPropagation(code);
+  EXPECT_TRUE(CompareShape(propagated_tsp, expected_tsp));
+}
+
+TEST(DataPropagationImplTest, SubTest) {
+  const char* code = R"ONNX(
+agraph (int32[2,4,6] x, int32[5] y) => (int32[3] w)
+{
+    xs = Shape(x)
+    ys = Shape(y)
+    z = Sub(xs, ys)
+    w = Cast<to = 7>(z)
+}
+)ONNX";
+  TensorShapeProto expected_tsp;
+  expected_tsp.mutable_dim()->Add()->set_dim_value(-3);
```

Comment:
-3 is not a valid value... This is an invalid model... can you change the input shapes so that that values are valid... 

invalid model test case will make sense if we have some logic to catch\handle such errors and we want to test that...